### PR TITLE
add: new field to getRoomData and update: getToken

### PR DESCRIPTION
**Pull request by HlainePoeEi**
### What You did

-   [x] add new field in getRoomData function
-   [x] update getToken function

### CODE REVIEW

File: app/Http/Controllers/api/v1/RoomController.php [hunk 1]
```diff
@@ -36,24 +36,33 @@ public function getToken(Request $request)
 
         $twilio = new Client($this->sid, $this->authToken);
 
-        $twilio->video->v1->rooms->create([
-            "uniqueName" => $request->room,
-        ]);
+        try {
+            $twilio->video->v1->rooms->create([
+                "uniqueName" => $request->room,
+                "type" => $request->type,
```
Comment (myonaingwinn) [inline app/Http/Controllers/api/v1/RoomController.php:42]:
```suggestion
                "type" => $request->type?: 'group',
```

**Review by myonaingwinn — COMMENTED** [review]
Onegai

**Review by myonaingwinn — APPROVED** [review]

File: app/Http/Controllers/api/v1/RoomController.php [hunk 2]
```diff
@@ -36,24 +36,33 @@ public function getToken(Request $request)
...
+                "MaxParticipants" => $request->participant_number,
+                "statusCallback" => $request->description,
+                "emptyRoomTimeout" => 1,
+                "maxParticipantDuration" => 60000,
```
Comment (myonaingwinn) [inline app/Http/Controllers/api/v1/RoomController.php:46]:
```suggestion
                "maxParticipantDuration" => 60000?:'',
```

**Review by myonaingwinn — DISMISSED** [review]
OK

File: app/Http/Controllers/api/v1/CustomObject/RoomData.php [hunk 3]
```diff
@@ -7,18 +7,18 @@ class RoomData
     public function getRoomData($twilio)
```
Comment (myonaingwinn) [inline app/Http/Controllers/api/v1/CustomObject/RoomData.php:7]: Please move this File to **Model** directory.

File: app/Http/Controllers/api/v1/CustomObject/RoomData.php [hunk 4]
```diff
@@ -7,18 +7,18 @@ class RoomData
...
             $record->name = $record->uniqueName;
             $record->id = $record->sid;
             $record->participant = count($participant);
             $record->maxParticipant = $record->maxParticipants;
+            $record->type = $record->type;
+            $record->description = $record->statusCallback;
+            $record->emptyRoomTimeout = $record->emptyRoomTimeout;
         }
 
         $result = (array('Room List' => $rooms));
```
Comment (myonaingwinn) [inline app/Http/Controllers/api/v1/CustomObject/RoomData.php:24]:
```suggestion
        $roomList = $rooms;
```

File: app/Http/Controllers/api/v1/RoomController.php [hunk 5]
```diff
@@ -36,10 +36,17 @@ public function getToken(Request $request)
...
+                "MaxParticipants" => (empty($request->participant_number)) ? '50' : $request->participant_number,
+                "statusCallback" => (empty($request->description)) ? 'null' : $request->description,
+                "emptyRoomTimeout" => (empty($request->empty_room_timeout)) ? '1' : $request->empty_room_timeout,
+            ]);
+        } catch (\Exception $e) {
+            echo $e->getMessage();
+        }
         $identity = $request->identity;
         $token = new AccessToken(
             $this->sid,
```
Comment (myonaingwinn) [inline app/Http/Controllers/api/v1/RoomController.php:52]:
```suggestion
            $this->sid,
            $this->apiKey,
            $this->apiSecret,
            3600,
            $identity
        );
        $grant = new VideoGrant();
        $grant->setRoom($request->room);
        $token->addGrant($grant);
        return ['token' => $token->toJWT()];
    }
    public function getRoomList()
    {
        $twilio = new Client($this->sid, $this->authToken);
        $obj = new RoomData();
        $roomList = $obj->getRoomData($twilio);
        return response()->json(["roomList" => $roomList], 200);
}
```

**Review by myonaingwinn — CHANGES_REQUESTED** [review]

File: app/Models/Room/RoomList.php [hunk 6]
```diff
@@ -1,27 +1,27 @@
 <?php
 
-namespace App\Http\Controllers\api\v1\CustomObject;
+namespace App\Models\Room;
 
-class RoomData
+class RoomList
 {
     public function getRoomData($twilio)
```
Comment (myonaingwinn) [inline app/Models/Room/RoomList.php:7]:
```suggestion
    public function getRoomList($twilio)
```

File: app/Http/Controllers/api/v1/RoomController.php [hunk 7]
```diff
@@ -60,9 +67,9 @@ public function getRoomList()
     {
         $twilio = new Client($this->sid, $this->authToken);
 
-        $obj = new RoomData();
-        $rooms_arr = $obj->getRoomData($twilio);
+        $obj = new RoomList();
+        $roomList = $obj->getRoomData($twilio);
```
Comment (myonaingwinn) [inline app/Http/Controllers/api/v1/RoomController.php:71]:
```suggestion
        $roomList = $obj->getRoomList($twilio);
```

**Review by myonaingwinn — CHANGES_REQUESTED** [review]

**Comment by myonaingwinn** [comment]
@HlainePoeEi お疲れ様！
Let's add pagination count!

File: app/Http/Controllers/api/v1/RoomController.php (hunk 7 above)
Comment (HlainePoeEi) [reply app/Http/Controllers/api/v1/RoomController.php:71]: controller htal ka getRoomList so tae function name chin same phit twar mhr moz e'dr ko update ma lote tot tar

Comment (myonaingwinn) [reply app/Http/Controllers/api/v1/RoomController.php:71]: 分かりました。有難うございます。

File: app/Models/Room/RoomList.php [hunk 8]
```diff
@@ -1,27 +1,27 @@
...
             $record->maxParticipant = $record->maxParticipants;
+            $record->type = $record->type;
+            $record->description = $record->statusCallback;
+            $record->emptyRoomTimeout = $record->emptyRoomTimeout;
         }
 
-        $result = (array('Room List' => $rooms));
+        $result = $rooms;
 
         return $result;
```
Comment (myonaingwinn) [inline app/Models/Room/RoomList.php:26]:
```suggestion
        return $rooms;
```

**Review by myonaingwinn — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/api/v1/RoomController.php [hunk 9]
```diff
@@ -36,10 +36,17 @@ public function getToken(Request $request)
...
 
-        $twilio->video->v1->rooms->create([
-            "uniqueName" => $request->room,
-        ]);
-
+        try {
+            $twilio->video->v1->rooms->create([
+                "uniqueName" => $request->room,
+                "type" => (empty($request->type)) ? 'group' : $request->type,
+                "MaxParticipants" => (empty($request->participant_number)) ? '50' : $request->participant_number,
```
Comment (myonaingwinn) [inline app/Http/Controllers/api/v1/RoomController.php]:
```suggestion
                "MaxParticipants" => (empty($request->maxParticipants)) ? '50' : $request->maxParticipants,
```

File: app/Http/Controllers/api/v1/RoomController.php [hunk 10]
```diff
@@ -36,10 +36,17 @@ public function getToken(Request $request)
...
+                "statusCallback" => (empty($request->description)) ? 'null' : $request->description,
+                "emptyRoomTimeout" => (empty($request->empty_room_timeout)) ? '1' : $request->empty_room_timeout,
```
Comment (myonaingwinn) [inline app/Http/Controllers/api/v1/RoomController.php:45]:
```suggestion
                "emptyRoomTimeout" => (empty($request->emptyRoomTimeout)) ? '1' : $request->emptyRoomTimeout,
```

File: app/Http/Controllers/api/v1/RoomController.php [hunk 11]
```diff
@@ -36,10 +36,17 @@ public function getToken(Request $request)
...
         $twilio = new Client($this->sid, $this->authToken);
 
-        $twilio->video->v1->rooms->create([
-            "uniqueName" => $request->room,
-        ]);
-
+        try {
+            $twilio->video->v1->rooms->create([
+                "uniqueName" => $request->room,
+                "type" => (empty($request->type)) ? 'group' : $request->type,
```
Comment (myonaingwinn) [inline app/Http/Controllers/api/v1/RoomController.php:42]:
```suggestion
                "type" => (empty($request->roomType)) ? 'group' : $request->roomType,
```

**Review by myonaingwinn — CHANGES_REQUESTED** [review]

**Review by myonaingwinn — APPROVED** [review]

**Review by myonaingwinn — APPROVED** [review]

**Comment by myonaingwinn** [comment]
Arigatouuuuuu

**Pull request merged**